kernel/driver.cc

```diff
@@ -498,6 +498,12 @@ int main(int argc, char **argv)
 	if (print_stats)
 		log_hasher = new SHA1;
 
+#if defined(__OpenBSD__)
+	// save the executable origin for proc_self_dirname()
```

Comment:
Best is to use extern defs here then in yosys.h.  just use same block here:
```
extern char *yosys_argv0;
extern char yosys_path[PATH_MAX];
```

Reply:
Do you mean to remind the reader that these variables are declared elsewhere?

I love the idea, but fail to add the declaration within `main()`:

```
[  0%] Building kernel/driver.o
kernel/driver.cc:502:22: error: definition or redeclaration of 'yosys_argv0' not allowed inside a function
        extern char *Yosys::yosys_argv0;
                     ~~~~~~~^
kernel/driver.cc:503:21: error: definition or redeclaration of 'yosys_path' not allowed inside a function
        extern char Yosys::yosys_path[PATH_MAX];
                    ~~~~~~~^
2 errors generated.
gmake: *** [Makefile:732: kernel/driver.o] Error 1
```

I can add it above main though:

```diff
--- a/kernel/driver.cc
+++ b/kernel/driver.cc
@@ -192,6 +192,11 @@ void yosys_atexit()
 #endif
 }

+#if defined(__OpenBSD__)
+extern char *Yosys::yosys_argv0;
+extern char Yosys::yosys_path[PATH_MAX];
+#endif
+
 int main(int argc, char **argv)
 {
        std::string frontend_command = "auto";
```

Sorry for the extra review for this!

Reply:
Yes, please do this change and also remove changes to yosys.h file